pkg/cron/cron.go

```diff
@@ -176,3 +200,30 @@ func (c *Cron) ValidateDomain(domain string) bool {
 	}
 	return false
 }
+
+// CheckConfigMap - check if config map exists in k8s cluster
+func (c *Cron) CheckConfigMap() bool {
+	configMap, err := c.k8sClient.CoreV1().ConfigMaps(configMapLoc).Get(configMapName, metav1.GetOptions{})
+	if err != nil || apiError.IsNotFound(err) {
+		return false
+	}
+	if configMap != nil {
+		return true
+	}
+	return false
+}
+
+// CreateConfigMap - create new config map in k8s cluster
+func (c *Cron) CreateConfigMap() *corev1.ConfigMap {
```

Comment:
I am thinking if we can merge everything into one CreateOrUpdateConfigMap function which takes in an etag and checks if the current configmap is on the cluster. This would encapsulate all the logic into here and just be function calls from the cron / controller.